File: java/src/main/java/com/tomoarrow/idv/bff/java/config/ApiClientConfig.java
package com.tomoarrow.idv.bff.java.config;

import com.tomoarrow.idv.client.generated.ApiClient;
import com.tomoarrow.idv.client.generated.api.DefaultApi;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ApiClientConfig {
    @Bean
    ApiClient idvApiClient(AppProperties properties) {
        ApiClient apiClient = new ApiClient();
        apiClient.updateBaseUri(properties.resolvedBaseUrl());
        return apiClient;
    }

    @Bean
    DefaultApi defaultApi(ApiClient apiClient) {
        return new DefaultApi(apiClient);
    }
}

File: kotlin/src/main/kotlin/com/tomoarrow/idv/bff/config/ApiClientConfig.kt
package com.tomoarrow.idv.bff.config

import com.tomoarrow.idv.client.apis.DefaultApi
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class ApiClientConfig(
    private val appProperties: AppProperties
) {

    @Bean
    fun defaultApi(): DefaultApi {
        val baseUrl = appProperties.idvBaseUrl.trimEnd('/')
        return DefaultApi(basePath = baseUrl)
    }
}
